### PR TITLE
timer decorator

### PR DESCRIPTION
**Pull request by jusschwitalla**
supply a python decorator which stops the time as needed by QUARK

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -202,3 +202,36 @@ def end_time_measurement(start: float) -> float:
     """
     end = time.perf_counter()
     return round((end - start) * 1000, 3)
+
+
+def quark_stop_watch(position: int = None):
```
Comment (Marvmann) [inline src/utils.py:207]: Please rename to "stop_watch"

File: src/utils.py [hunk 2]
```diff
@@ -202,3 +202,36 @@ def end_time_measurement(start: float) -> float:
...
+    results in valid:    
+    ```
+    processed_data, time_to_process = run(input,...)
+    ```
+    If the return value of the decorated function is of type tuple,
+    the optional parameter `position` can be used to specify the position at which the
+    measured time is to be inserted in the returned tuple.
+
+    :param position: the position at which to insert the time
+    :type position: int
```
Comment (Marvmann) [inline src/utils.py:224]: Please add a parameter and type description for the output as well

File: src/utils.py [hunk 3]
```diff
@@ -202,3 +202,36 @@ def end_time_measurement(start: float) -> float:
...
+    ```
+    results in valid:    
+    ```
+    processed_data, time_to_process = run(input,...)
+    ```
+    If the return value of the decorated function is of type tuple,
+    the optional parameter `position` can be used to specify the position at which the
+    measured time is to be inserted in the returned tuple.
+
+    :param position: the position at which to insert the time
```
Comment (Marvmann) [inline src/utils.py:223]: Please clarify, e.g., The position at which measured time is returned in the output tuple

**Review by Marvmann — CHANGES_REQUESTED** [review]
Added a couple of comments.

File: src/utils.py (hunk 2 above)
Comment (jusschwitalla) [reply src/utils.py:225]: done. But note that the user of the decorator is usually not interested in the return value of the decorator but in the return value of the decorated function.

File: src/utils.py (hunk 1 above)
Comment (jusschwitalla) [reply src/utils.py]: done

File: src/utils.py (hunk 3 above)
Comment (jusschwitalla) [reply src/utils.py]: done

**Review by Marvmann — APPROVED** [review]
Thanks for the changes!

**Pull request merged**